ui/raidboss/data/03-hw/dungeon/gubal_library_hard.js

```diff
@@ -0,0 +1,300 @@
+        if (data.role == 'tank') {
+          return {
+            en: 'Tank Cleave on you',
+          };
+        }
+        return {
+          en: 'Avoid tank cleave',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Darkness',
+      regex: /Darkness \(buster\)/,
+      beforeSeconds: 5,
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank';
+      },
+      infoText: {
+        en: 'Tank Buster',
+      },
+    },
+  ],
+  triggers: [
+    {
+      id: 'Gubal Hard Bibliocide',
+      regex: / 14:1945:Liquid Flame starts using Bibliocide/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      infoText: {
+        en: 'AoE',
+      },
+    },
+    {
+      // Ferrofluid handling is shamelessly copied from Hades Normal
+      id: 'Gubal Hard Ferrofluid Collect',
+      regex: / 1B:\y{ObjectId}:(\y{Name}):....:....:(0030|0031):/,
+      run: function(data, matches) {
+        data.fluid = data.fluid || {};
+        if (matches[2] == '0030')
+          data.fluid[matches[1]] = 'positive';
+        if (matches[2] == '0031')
+          data.fluid[matches[1]] = 'negative';
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Execute',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:(?:0030|0031):/,
+      suppressSeconds: 5,
+      delaySeconds: 0.5,
+      infoText: function(data) {
+        if (data.fluid[data.me] == data.fluid['Liquid Flame']) {
+          return {
+            en: 'Close to boss',
+          };
+        }
+        return {
+          en: 'Away from boss',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Cleanup',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:003[01]:/,
+      delaySeconds: 5,
+      run: function(data) {
+        delete data.fluid;
+      },
+    },
+    {
+      id: 'Gubal Hard Slosh',
+      regex: / 23:\y{ObjectId}:Liquid Flame:\y{ObjectId}:(\y{Name}):....:....:0039/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: {
+        en: 'Away from boss',
+      },
+    },
+    {
+      id: 'Gubal Hard Seals',
+      regex: / 1A:\y{ObjectId}:(\y{Name}) gains the effect of (Moon|Sun)seal/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: function(data, matches) {
+        if (matches[2] == 'Moon') {
+          return {
+            en: 'Stand in blue',
+          };
+        }
+        return {
+          en: 'Stand in red',
+        };
+      },
+    },
+    {
+      // This inflicts a vulnerability stack on the tank if not interrupted
+      id: 'Gubal Hard Condensed Libra',
+      regex: / 14:198D:Mechanoscribe starts using Condensed Libra on \y{Name}/,
+      infoText: function(data) {
+        if (data.CanStun()) {
+          return {
+            en: 'Stun Mechanoscribe',
+          };
+        }
+        if (data.CanSilence()) {
```

Comment:
Can you put the silence first before the stun? Sometimes things get stun resistant, so if you can silence, I think that's preferred.

Reply:
Good point, I hadn't considered that.